modules/pam_namespace/pam_namespace.c

```diff
@@ -707,14 +795,17 @@ static int parse_config_file(struct instance_data *idata)
 	fclose(fil);
 	free(line);
 
-	if (n >= globbuf.gl_pathc)
+	if (filename_list == NULL || filename_list[n] == NULL)
 	    break;
 
-	confname = globbuf.gl_pathv[n];
-	n++;
+	confname = filename_list[n++];
+    }
+    if (filename_list != NULL) {
+        for (int i = 0; filename_list[i] != NULL; i++)
```

Comment:
`i` should be of the same type here as `i` in the function where the array is initialized.

Reply:
fixed